Problem2.java

```diff
@@ -0,0 +1,88 @@
+/**
+LeetCode Submitted : YES
+Time Complexity : O(Num of words * Max Height of Tree)
+Space Complexity : O(Num of words)
+
```

Comment:
Space complexity will be O(n*k) where n is number of words in list to be inserted and k is length of longest word in dictionary.